Remove "experimental_capture" kwarg from "qjit" (#1657)
PennyLaneAI/catalyst: doc/releases/changelog-dev.md

<h3>Breaking changes 💔</h3>

* Catalyst has removed the `experimental_capture` keyword from the `qjit` decorator in favour of
  unified behaviour with PennyLane. 
  [(#1657)](https://github.com/PennyLaneAI/catalyst/pull/1657)

  Instead of enabling program capture with Catalyst via `qjit(experimental_capture=True)`, program capture

>   can be enabled via the global toggle `qml.capture.enable`:

Reviewer: ```suggestion
  can be enabled via the global toggle `qml.capture.enable()`:
```

  
  ```python
  import pennylane as qml
  from catalyst import qjit

  dev = qml.device("lightning.qubit", wires=2)

  qml.capture.enable()

  @qjit
  @qml.qnode(dev)
  def circuit(x):
      qml.Hadamard(0)
      qml.CNOT([0, 1])
      return qml.expval(qml.Z(0))

  circuit(0.1)

>   ```

Reviewer: ````suggestion
  ```

  Disabling program capture can be done with `qml.capture.disable()`.
````

<h3>Breaking changes 💔</h3>

> * Catalyst has removed the `experimental_capture` keyword from its `qjit` function in favor of a
>   unified program capture behaviour. 
>   [(#1657)](https://github.com/PennyLaneAI/catalyst/pull/1657)
> 
>   Program capture has to be enabled before the definition of the function to be qjitted, and 
>   should be disabled after execution.

Reviewer: ```suggestion
* Catalyst has removed the `experimental_capture` keyword from the `qjit` decorator in favour of
  unified behaviour with PennyLane. 
  [(#1657)](https://github.com/PennyLaneAI/catalyst/pull/1657)

  Instead of enabling program capture with Catalyst via `qjit(experimental_capture=True)`, program capture
  can be enabled via the global toggle `qml.capture.enable`:
```

@rauletorresc curious why capture has to be disabled at the end of a program? 🤔

Reviewer 2: same

<h3>Breaking changes 💔</h3>

* Catalyst has removed the `experimental_capture` keyword from its `qjit` function in favor of a
  unified program capture behaviour. 
  [(#1657)](https://github.com/PennyLaneAI/catalyst/pull/1657)

  Program capture has to be enabled before the definition of the function to be qjitted.
  
  For AOT compilation, program capture can be disabled right after the qjit usage and before execution.
  
  ```python
  import pennylane as qml
  from catalyst import qjit

  dev = qml.device("lightning.qubit", wires=1)

  qml.capture.enable()

>   @qjit()

Reviewer 2: ```suggestion
  @qjit
```

Author: Done

  @qml.qnode(dev)
  def circuit(x: float):
      qml.Hadamard(0)
      qml.CNOT([0, 1])
      return qml.expval(qml.Z(0))

  qml.capture.disable()

>   circuit(0.1)

Reviewer 2: I don't know if we should be promoting this pattern at all, what is the point of disabling it before execution?

Author: Removed.
